src/shared/ui/button/Button.style.ts

```diff
@@ -1,65 +1,147 @@
+  gap: ${icon ? '0.4rem' : '0'};
 `;
 
-export const StyledButton = styled.button<{ icon?: ReactElement }>`
-  display: inline-flex;
-  align-items: center;
-  justify-content: center;
-  ${({ icon }) => paddingStyles(icon)};
-  background-color: ${({ theme }) => theme.color.PRIMARY_BLUE};
-  color: ${({ theme }) => theme.color.WHITE};
-  border: 0.1rem solid transparent;
-  border-radius: 0.8rem;
-  cursor: pointer;
-  box-sizing: border-box;
-  flex-shrink: 0;
-  ${typo.BODY_SEMIBOLD};
-
+const commonHoverStyles = css`
   &:hover {
+    border: 0.08rem solid ${({ theme }) => theme.color.GRAY_30};
     background-color: ${({ theme }) => theme.color.BLUE_30};
-    color: ${({ theme }) => theme.color.PRIMARY_BLUE};
-    border: 0.1rem solid ${({ theme }) => theme.color.PRIMARY_BLUE};
   }
+`;
 
+const commonActiveStyles = css`
   &:active {
-    background-color: ${({ theme }) => theme.color.WHITE};
-    color: ${({ theme }) => theme.color.PRIMARY_BLUE};
-    border: 0.1rem solid ${({ theme }) => theme.color.PRIMARY_BLUE};
-  }
-
-  &:disabled {
-    background-color: ${({ theme }) => theme.color.BLUE_30};
-    color: ${({ theme }) => theme.color.GRAY_50};
-    border: 0.1rem solid ${({ theme }) => theme.color.BLUE_50};
-    cursor: not-allowed;
+    background-color: ${({ theme }) => theme.color.PRIMARY_BLUE};
+    color: ${({ theme }) => theme.color.WHITE};
   }
 `;
 
-export const IconWrapper = styled.div<{ iconType?: 'bookmark' | 'list' }>`
-  display: inline-flex;
-  margin-right: 0.4rem;
-  width: 2.2rem;
-  height: 2.2rem;
+export const buttonStyles = css`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border: none;
+  white-space: nowrap;
+  cursor: pointer;
 
   svg {
-    width: 100%;
-    height: 100%;
+    width: 2.2rem;
+    height: 2.2rem;
+    display: block;
+    margin: auto;
   }
+`;
 
-  ${StyledButton}:hover & svg {
-    fill: ${({ iconType, theme }) => (iconType === 'list' ? theme.color.PRIMARY_BLUE : 'none')};
-    stroke: ${({ iconType, theme }) =>
-      iconType === 'bookmark' ? theme.color.PRIMARY_BLUE : 'none'};
-  }
+export const variantStyles = (variant: Required<ButtonProps>['variant']) => {
+  const style = {
+    primary: css<{ icon?: React.ReactElement }>`
+      ${typo.BODY_SEMIBOLD};
+      border: 0.1rem solid transparent;
+      border-radius: 0.8rem;
+      background-color: ${({ theme }) => theme.color.PRIMARY_BLUE};
+      color: ${({ theme }) => theme.color.WHITE};
+      ${({ icon }) => getPaddingWithIcon(icon)}
 
-  ${StyledButton}:active & svg {
-    fill: ${({ iconType, theme }) => theme.color.PRIMARY_BLUE};
-    stroke: ${({ iconType, theme }) =>
-      iconType === 'bookmark' ? theme.color.PRIMARY_BLUE : 'none'};
-  }
+      &:hover {
+        background-color: ${({ theme }) => theme.color.BLUE_30};
+        color: ${({ theme }) => theme.color.PRIMARY_BLUE};
+        border: 0.1rem solid ${({ theme }) => theme.color.PRIMARY_BLUE};
```

Comment:
```suggestion
        border-color : ${({ theme }) => theme.color.PRIMARY_BLUE};
```
색상만 바뀌는거니 위와 같이 하는건 어떨까요!?

Reply:
왓 놓치고 있었던 부분이네요 ㅠㅠ 해당 방식이 더 깔끔하게 적용될 거 같습니다~

Reply:
[1c9f4b2](https://github.com/GrebbMe/grabbme-fe/pull/57/commits/1c9f4b2fa85d0c34477e10ef4b6f2750b20a2e9d) 에 반영하였습니다!